packages/ui/fe-bundle/src/utils/theme/loadDocsTypography.ts

```diff
@@ -0,0 +1,122 @@
+import type * as FernRegistryDocsRead from "@fern-fern/registry-browser/api/resources/docs/resources/v1/resources/read";
+
+export function getFontExtension(url: URL): string {
+    const ext = url.pathname.split(".").pop();
+    if (ext == null) {
+        throw new Error("No extension found");
+    }
+    return ext;
+}
+
+export type FontType = "headingsFont" | "bodyFont" | "codeFont";
+
+export interface GenerationFontConfig {
+    fontType: FontType;
+    fontName: string;
+    fontExtension: string;
+    fontUrl: FernRegistryDocsRead.Url | undefined;
```

Comment:
Let's make this required and throw below if `docsDefinition.files[typographyFontType.fontFile]` is undefined